Update library to 3.10.0 generate compiling error on ESP32

## Overview

__Actual behavior:__


Compiling error:

```cpp
Compiling .pio/build/TTGO_T7/lib074/Update/HttpsOTAUpdate.cpp.o
In file included from .pio/libdeps/TTGO_T7/ESP8266 Influxdb/src/query/FluxTypes.cpp:28:0:
.pio/libdeps/TTGO_T7/ESP8266 Influxdb/src/query/FluxTypes.h: In constructor 'FluxString::FluxString(String, String, const char*)':
.pio/libdeps/TTGO_T7/ESP8266 Influxdb/src/query/FluxTypes.h:131:12: error: 'FluxString::value' will be initialized after [-Werror=reorder]
     String value;
            ^
.pio/libdeps/TTGO_T7/ESP8266 Influxdb/src/query/FluxTypes.h:127:17: error:   'const char* FluxString::_type' [-Werror=reorder]
     const char *_type;
                 ^
.pio/libdeps/TTGO_T7/ESP8266 Influxdb/src/query/FluxTypes.cpp:147:1: error:   when initialized here [-Werror=reorder]
 FluxString::FluxString(String rawValue, String value, const char *type):FluxBase(rawValue),value(value),_type(type)
 ^
Compiling .pio/build/TTGO_T7/lib074/Update/Updater.cpp.o
Compiling .pio/build/TTGO_T7/libc61/FS/FS.cpp.o
```

__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Updated dependencie from 3.9.0 to 3.10.0
2. Compiling over ESP32 Arduino Framework using PlatformIO

__Expected behavior:__

Backward compatibility

__Specifications:__
 - Library Version: 3.10.0
 - InfluxDB Version: NA
 - Board/ESP chip: ESP32 (lolin32) 
 - Device Arduino SDK version: Espressif 32 3.4.0

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@
   - Arduino `String` class
   - C `char *` or `char[]` 
   - Flash string using `F`,`PSTR` or `FPSTR` macros
+### Fixes 
+ - [176](https://github.com/tobiasschuerg/InfluxDB-Client-for-Arduino/pull/176) - Cleared all compiler warnings
 
 ## 3.10.0 [2022-01-20]
 ### Features

--- a/src/InfluxDbClient.cpp
+++ b/src/InfluxDbClient.cpp
@@ -681,8 +681,8 @@ InfluxDBClient::BatchStreamer::BatchStreamer(InfluxDBClient::Batch *batch) {
     _length = 0;
     _pointer = 0;
     _linePointer = 0;
-    for(int i=0;i<_batch->pointer;i++) {
-    _length += strlen(_batch->buffer[i])+1;
+    for(uint16_t i=0;i<_batch->pointer;i++) {
+        _length += strlen(_batch->buffer[i])+1;
     }
 }
 
@@ -703,8 +703,8 @@ int InfluxDBClient::BatchStreamer::read(uint8_t* buffer, size_t len) {
 size_t InfluxDBClient::BatchStreamer::readBytes(char* buffer, size_t len) {
 
     INFLUXDB_CLIENT_DEBUG("BatchStream::readBytes %d\n", len);
-    int r=0;
-    for(int i=0;i<len;i++) {
+    unsigned int r=0;
+    for(unsigned int i=0;i<len;i++) {
         if(available()) {
             buffer[i] = read();
             r++;
@@ -740,12 +740,9 @@ int InfluxDBClient::BatchStreamer::peek() {
     } else {
         r = _batch->buffer[_pointer][_linePointer];
     }
-    // #ifdef INFLUXDB_CLIENT_DEBUG_ENABLE
-    //     Serial.printf_P(PSTR("%c"), r);
-    // #endif
     return r;
 }
 
-size_t InfluxDBClient::BatchStreamer::write(uint8_t data)  {
+size_t InfluxDBClient::BatchStreamer::write(uint8_t)  {
     return 0;
 }

--- a/src/query/FluxTypes.cpp
+++ b/src/query/FluxTypes.cpp
@@ -38,8 +38,7 @@ const char	*FluxBinaryDataTypeBase64         = "base64Binary";
 const char	*FluxDatatypeDatetimeRFC3339      = "dateTime:RFC3339";
 const char	*FluxDatatypeDatetimeRFC3339Nano  = "dateTime:RFC3339Nano";
 
-FluxBase::FluxBase(const String &rawValue) {
-    _rawValue = rawValue;
+FluxBase::FluxBase(const String &rawValue):_rawValue(rawValue) {
 }
 
 FluxBase::~FluxBase() {
@@ -144,7 +143,7 @@ FluxString::FluxString(const String &rawValue, const char *type):FluxString(rawV
 
 }
 
-FluxString::FluxString(const String &rawValue, const String &value, const char *type):FluxBase(rawValue),value(value),_type(type)
+FluxString::FluxString(const String &rawValue, const String &value, const char *type):FluxBase(rawValue),_type(type),value(value)
 {
 
 }

--- a/src/query/Params.cpp
+++ b/src/query/Params.cpp
@@ -52,7 +52,7 @@ QueryParams &QueryParams::operator=(const QueryParams &other) {
 }
 
 QueryParams &QueryParams::add(const String &name, float value, int decimalPlaces) {
-  return add(name, (double)value);
+  return add(name, (double)value, decimalPlaces);
 }
 
 QueryParams &QueryParams::add(const String &name, double value, int decimalPlaces) {

